Functions/xASL_Iteration.m

```diff
@@ -41,65 +48,80 @@
 		dbSettings.settings.dryRun = dryRun;
 	end
 	
-	% Set the stopAfterErrors field
-	if nargin < 4 || isempty(stopAfterErrors)
-		if ~isfield(x.settings,'stopAfterErrors') || isempty(x.settings.stopAfterErrors)
-			dbSettings.settings.stopAfterErrors = Inf;
-		else
-			dbSettings.settings.stopAfterErrors = x.settings.stopAfterErrors;
-		end
-	else
-		dbSettings.settings.stopAfterErrors = stopAfterErrors;
+    % Set the stopAfterErrors field
+    if nargin < 4 || isempty(stopAfterErrors)
+        if ~isfield(x.settings,'stopAfterErrors') || isempty(x.settings.stopAfterErrors)
+            dbSettings.settings.stopAfterErrors = Inf;
+        else
+            dbSettings.settings.stopAfterErrors = x.settings.stopAfterErrors;
+        end
+    else
+        dbSettings.settings.stopAfterErrors = stopAfterErrors;
     end
-	    
+    
+    % Check if it was possible to load the subjects
     if nargin<5 || isempty(SelectedSubjects)
-        try
+        if isfield(x,'SUBJECTS')
```

Comment:
I agree. Lets put it back here and remove from the Import scripts. Michael agrees?

Reply:
Oh yes, sorry about that, we can definitely use the previous version regarding [this diff](https://github.com/ExploreASL/ExploreASL/commit/5a7034ba82929c81b85be3dedd53ec58b2d6e476#diff-dbe9affae9c7d0938ce3c68c6890f26d105b43cc9dc4239b4c16e16e91041d57). Not 100% sure why I changed that. In general the whole `xASL_Iterate` script is run using a try-catch, so generally it shouldn't break the pipeline anyway, but the remaining script behavior is somewhat unclear. It would probably be correct to at least write it like this...

```matlab
    if nargin<5 || isempty(SelectedSubjects)
        try
            SelectedSubjects = x.SUBJECTS;
        catch ME
            warning('Loading subjects didnt work, ...');
            fprintf('%s\n', ME.message);
            SelectedSubjects = [ ];
        end
    end
```

... instead of this ...

```matlab
    if nargin<5 || isempty(SelectedSubjects)
        try
            SelectedSubjects = x.SUBJECTS;
        catch ME
            warning('Loading subjects didnt work, ...');
            fprintf('%s\n', ME.message);
        end
    end
```

... so with some kind of default instead of just leaving the whole thing empty, but it probably worked for most of the cases before, so it's more of a code style thing.

Reply:
This comes down to one of our major problems. There are a lot of parts in the code where a structure (`x`) goes in and then some fields are added **conditionally** without basic defaults. In these cases it can be helpful to write that into the description (e.g. this goes in, this is required, this goes out), but I know that this is a ton of work.

Reply:
Fix ea03421 ?